vllm/v1/sample/metadata.py

```diff
@@ -37,3 +37,11 @@ class SamplingMetadata:
     min_tokens: Dict[int, Tuple[int, Set[int]]]
 
     logit_bias: List[Optional[Dict[int, float]]]
+
+    # These two parameters are for allowed_token_ids.
+    # `no_allowed_token_ids`` is a bool to indicate whether we have
+    # allowed_token_ids.
+    # `allowed_token_ids_mask` is a 2D bool tensor of shape (max batch size,
+    # vocab size).
+    no_allowed_token_ids: bool
+    allowed_token_ids_mask: Optional[torch.Tensor]
```

Comment:
Actually I don't think we need the bool since that's what `None` means? (similar to how `top_k`, `top_p`, etc. are now).

Reply:
There are cases, if allowed_token_ids_mask got initialized, it will not go back again. So I am still trying to keep the bool to indicate whether we indeed need to apply allowed_token_ids.

Reply:
Right but the bool isn't needed for that in the metadata. You can set it dependent on the `no_allowed_token_ids` property (whether the set is empty) ... same as the top_p etc. fields in here if you look at where they're set. That's independent of the lazy loading.